lib/SILOptimizer/Mandatory/TFPartition.cpp

```diff
@@ -622,6 +624,7 @@ class TFFunctionPartition {
   DenseMap<StringRef, std::unique_ptr<LoweredGraphFunction>> &graphFunctions;
   GraphFunctionDeviceInfo deviceInfo; // Device placement info.
   DominanceInfo &DI;
+  SILLoopInfo &LI;
```

Comment:
This shouldn't need LoopInfo.  You can just pass a null pointer into splitAllCondBrCriticalEdgesWithNonTrivialArgs

Reply:
Addressed in https://github.com/apple/swift/pull/18437.